packages/steveo/src/types/workflow.ts

```diff
@@ -5,7 +5,7 @@ import { TaskOptions } from './task-options';
  * related message (either workflow or workflow step).
  */
 export interface WorkflowPayload {
-  workflowId?: string;
+  workflowId: string | undefined;
```

Comment:
@treefitty dummy question. What's the difference between the 2 approaches here?

Reply:
no worries at all @lmagalhaes ....

when you use `?` you are saying the parameter is optional, it can be left out when calling the function. the corollary is that all proceeding arguments must also be optional (i.e. have the `?`)

when you declare `string | undefined` you are saying "it can be a string, or the literal value undefined.... but you must explicitly provide one or the other!"

Reply:
@treefitty Right. That’s what I thought. But what’s the advantage on enforcing it to be undefined if not providing any value would also result in undefined?
By making this change we are forcing changes in all the classes that implements that interface as well as every code that uses the concrete implementation.
I don’t know in how often this workflow payload is used but we may end-up with lots os changes.

Reply:
There are currently no places this is being used, it's new code that is entering production use, so it was a consideration that I'd prefer it to be explicitly provided while making that move toward prod completeness. Generally I only use optional parameters for extremely trivial values and prefer callers to be explicit .... this is obviously a generalisation  😅